lib/facebot.js

```diff
@@ -154,11 +155,41 @@ Facebot.prototype.createFBApi = function(credentials){
            });
 }
 
+// loop that will continue to send the is typing indicator to a channel
+// until we hear back they are not typing, or 10 minutes have past
+Facebot.prototype.typingLoop = function(link){
+    if (! link.is_typing || ((Date.now() / 1000) - link.typing_start_time) > 60*10) //if they were typing for more than 10 minutes lets assume we missed the typing end event
+        return;
+    var us = this;
```

Comment:
Been awhile since doing this JS stuff, hate most of it although the Q library does make it better.  Cannot wait for proper async/await as it will be so much cleaner.  Right now sometimes my this was losing context so I sometimes am better safe than sorry will update.